rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -331,7 +331,28 @@ public void setLastRetiredFederationP2SHScript(Script lastRetiredFederationP2SHS
     @Override
     public void setSvpFundTransactionUnsignedHash(Sha256Hash hash) {
         this.svpFundTransactionUnsignedHash = hash;
-        isSvpFundTransactionUnsignedHashSet = true;
+        this.isSvpFundTransactionUnsignedHashSet = true;
+    }
+
+    @Override
+    public Optional<Sha256Hash> getSvpFundTransactionUnsignedHash(ActivationConfig.ForBlock activations) {
+
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (svpFundTransactionUnsignedHash != null) {
+            return Optional.of(svpFundTransactionUnsignedHash);
+        }
+
+        if (isSvpFundTransactionUnsignedHashSet) {
```

Comment:
I'd add a comment here explaining why it returns empty in this case

Reply:
agree

Reply:
Updated.